src/components/Aframe/my-loader.js

```diff
@@ -432,144 +394,91 @@ AFRAME.registerComponent("myloader", {
-          material.uniforms.useLut.value = false;
-          material.uniforms.channel.value = this.data.channel;
-          material.needsUpdate = true;
-        }
-      } else {
-        this.updateColorMapping();
-      }
+      this.updateOpacityData(
+        this.data.alphaXDataArray,
+        this.data.alphaYDataArray
+      );
+      this.updateTransferTexture();
     }
 
-    if (this.colorMapEnabled) {
-      // this part updates the opacity control points
-      if (
-        (this.data.alphaXDataArray !== undefined &&
-          oldData.alphaXDataArray !== this.data.alphaXDataArray) ||
-        (this.data.alphaYDataArray !== undefined &&
-          oldData.alphaYDataArray !== this.data.alphaYDataArray)
-      ) {
-        this.newAlphaData = [];
-
-        for (var i = 0; i <= this.data.alphaXDataArray.length - 2; i++) {
-          var scaledColorInit = this.data.alphaXDataArray[i] * 255;
-          var scaledColorEnd = this.data.alphaXDataArray[i + 1] * 255;
-
-          var scaledAplhaInit = this.data.alphaYDataArray[i] * 255;
-          var scaledAlphaEnd = this.data.alphaYDataArray[i + 1] * 255;
-
-          var deltaX = scaledColorEnd - scaledColorInit;
-
-          for (var j = 1 / deltaX; j < 1; j += 1 / deltaX) {
-            // linear interpolation
-            this.newAlphaData.push(
-              scaledAplhaInit * (1 - j) + scaledAlphaEnd * j
-            );
-          }
-        }
-
-        this.updateTransfertexture();
-      }
-
-      if (
-        this.colorMapEnabled &&
-        oldData.colorMap !== undefined &&
-        oldData.colorMap !== this.data.colorMap
-      ) {
-        if (this.data.transferFunction) {
-          var imgColorImage = document.querySelector(".colorMapImg");
-          var imgWidth = imgColorImage.width;
-          var imgHeight = imgColorImage.height;
-
-          var colorCanvas = document.createElement("canvas");
-          // var el = this.el;
-
-          // var alpha = this.alphaData;
-          var colorTransfer = this.colorTransfer;
-          var iam = this;
-          this.colorMap.img.onload = function (data) {
-            colorCanvas.height = imgHeight;
-            colorCanvas.width = imgWidth;
-            var colorContext = colorCanvas.getContext("2d");
-            colorContext.drawImage(imgColorImage, 0, 0);
-            var colorData = colorContext.getImageData(0, 0, imgWidth, 1).data;
-            colorTransfer = new Uint8Array(3 * 256);
-            for (var i = 0; i < 256; i++) {
-              colorTransfer[i * 3] = colorData[i * 4];
-              colorTransfer[i * 3 + 1] = colorData[i * 4 + 1];
-              colorTransfer[i * 3 + 2] = colorData[i * 4 + 2];
-            }
-            iam.colorTransfer = colorTransfer;
-            iam.updateTransfertexture();
-          };
-          this.colorMap.img.src = imgColorImage.src;
-        }
-      }
+    if (oldData.colorMap !== this.data.colorMap) {
+      this.currentColorMap = this.data.colorMap;
```

Comment:
should this call `updateColorMap`?

Reply:
I missed it. I'll revisit it after Robert's merge